clientv3/kv.go

```diff
@@ -106,6 +118,30 @@ func (kv *kv) Delete(key string) (*DeleteResponse, error) {
 	return (*DeleteResponse)(r.GetResponseDeleteRange()), nil
 }
 
+func (kv *kv) Compact(rev int64) error {
+	for {
+		r := &pb.CompactionRequest{Revision: rev}
+		_, err := kv.remote.Compact(context.TODO(), r)
+		if err == nil {
+			return nil
+		}
+
+		if isRPCError(err) {
```

Comment:
not really. if this is a RPC error, we should return the error directly. Or this is a network error, then we try to switchRemote etcd.